packages/core/src/BinaryData/FileSystem.manager.ts

```diff
@@ -112,23 +134,31 @@ export class FileSystemManager implements BinaryData.Manager {
 		const oldPath = this.resolvePath(oldFileId);
 		const newPath = this.resolvePath(newFileId);
 
-		await Promise.all([
-			fs.rename(oldPath, newPath),
-			fs.rename(`${oldPath}.metadata`, `${newPath}.metadata`),
-		]);
+		await assertDir(path.dirname(newPath));
+
+		await fs.rename(oldPath, newPath);
+		await fs.rename(`${oldPath}.metadata`, `${newPath}.metadata`);
```

Comment:
Is there a reason you took these our of a `Promise.all`?

Reply:
Unintended during refactoring, restored.